java/src/jmri/script/swing/ScriptEngineSelector.java

```diff
@@ -53,15 +53,10 @@ public static ScriptEngineSelector getScriptEngineSelector() {
     /**
      * Get the currently selected engine
      */
-    @CheckForNull
-    public ScriptEngine getEngine() {
+    @Nonnull
+    public ScriptEngine getEngine() throws ScriptException {
         String language = languageIDs.get(getSelectedIndex());
```

Comment:
Might this cause problems on Java 15/17 if Graal is not installed? What happens if there is no script engine installed at all? Or do we either require that users of Java 15/17 have GraalVM installed or that we have resolved how to ship Graal with JMRI at that point?

There is a document describing how to ship Graal with JMRI to be able to run Graal on a stock JDK.
https://medium.com/graalvm/graalvms-javascript-engine-on-jdk11-with-high-performance-3e79f968a819

And it has an example project:
https://github.com/graalvm/graal-js-jdk11-maven-demo

But I don't know enough of Maven POM files to understand how to copy that to JMRI. And maybe we doesn't even need to try, if we decide to have GraalVM JDK required for JMRI or even ship JMRI precompiled with GraalVM.

Reply:
Thanks for checkin. A couple points:
 - The change is to throw a ScriptException instead of returning null; either one's an error
 - The engine that wraps Python 3 signals that it can't run via the getVersion() method.  That returns a null if there's no working engine of that type (name and language).  That in turn is checked in ScriptEngineSelector early on.  So a problem at this late stage isn't due to missing Graal (or other component), but something more unexpected. 

As to Java15/17 time, I don't think anything is decided about how to handle that.  I'm not sure how many people are using ECMAScript (Is it used by the web server? I don't know), and that's the only thing for sure lost.  We could recommend people install GraalVM to get it back.  Or we could build it, following the links you found. Or, if we can get native-image to work by then, we could distribute native images with the script engines included. In that case, not JRE/JDK/VM would even need to be installed.